English Altibase_7_1_0_9_3_Patch_Notes.md (#517)
ALTIBASE/Documents: PatchNotes/Altibase_7.1/eng/Altibase_7_1_0_9_3_Patch_Notes.md

# Altibase 7.1.0.9.3 Patch Notes

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

- [New Features](#new-features)

>     - [BUG-50713 Improve that the error message includes the header information when the Protocol header error occurs because of an incorrect packet receiving.](#bug-50713)

Reviewer: due to malformed packet 이 더 맞을 것 같습니다.

>     - [BUG-50713 Improve that the error message includes the header information when the Protocol header error occurs because of an incorrect packet receiving.](#bug-50713)

Author: ```suggestion
    - [BUG-50713 Improve that the error message includes the header information when the Protocol header error occurs due to malformed packets.](#bug-50713)
```

- [Fixed Bugs](#fixed-bugs)
    - [BUG-48041 Fix an issue where triggers using %ROWTYPE or %TYPE did not automatically recompile when the table definition changed.](#bug-48041)
    - [BUG-50694 Resolve a problem where the server would abnormally terminate when a group by column in a subquery within a disk table was referenced in the main query.](#bug-50694)
    - [BUG-50710 Fix an issue where the server could abnormally terminate when creating a package that calls an external procedure.](#bug-50710)
    - [BUG-50714  Fix a problem that Altibase does not work properly when directly modifying a variable with the constant or nocopy option within a package.](#bug-50714)
    - [BUG-50729 Address an issue where the stack size of terminated threads continued to be output in V$MEMSTAT.](#bug-50729)
    - [BUG-50758 Fix an escape processing omission for double quote characters in APRE.](#bug-50758)
    - [BUG-50778 Add exception handling to the getColumnCount function.](#bug-50778)

>     - [BUG-50783 Resolve an issue where an "Invalid use of host variables" error occurred during DDL on a table while executing a procedure.](#bug-50783)

Author: ```suggestion
    - [BUG-50783 Resolve the issue where "ERR-3123B : Invalid use of host variables" occurred and a procedure was terminated when a DDL statement was executed on the table being referenced within the procedure.](#bug-50783)
```

    - [BUG-50789 Fix a problem where media recovery failed when additional checkpoint image files were added during the process.](#bug-50789)

>     - [BUG-50792 Fix a logic error in calculating the maximum value of undo tablespace during snapshot specification.](#bug-50792)

Author: ```suggestion
    - [BUG-50792 Fix a logic error in calculating the maximum value of undo tablespace when SNAPSHOT is set up.](#bug-50792)
```

- [Changes](#changes)
    - [Version Info](#version-info)
    - [Compatibility](#Compatibility)
    - [Altibase Server Properties](#Altibase-Server-Properties)
    - [Performance Views](#Performance-Views)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

New Features
============

> ### BUG-50713<a name=bug-50713></a> Improve that the error message includes the header information when the Protocol header error occurs because of an incorrect packet receiving.

Reviewer: ```suggestion
### BUG-50713<a name=bug-50713></a> Improve that the error message includes the header information when the Protocol header error occurs due to malformed packets.
```

-   **module** : cm
-   **Category** : Functionality
-   **Reproducibility** : Rare

> -   **Description** : When the Protocol header error occurs because of an incorrect packet receiving, the error message now displays the header information to assist in problem analysis.

Author: ```suggestion
-   **Description** : When the Protocol header error occurs due to malformed packets, the error message now displays the header information to assist in problem analysis.
```

        ulpSqlstmt.ismt = 0;
        ulpSqlstmt.numofhostvar = 0;
        ulpSqlstmt.statusptr = NULL;
        ulpSqlstmt.errcodeptr = NULL;
        ulpSqlstmt.isatomic = 0;
        ulpSqlstmt.stmt = (char *)"SELECT ' " ' FROM DUAL";
     ...
    }
    }
    ```

  -   **Expected Results**

      In the .c file, users can find \ is added before " as follows.
      
      ```c
      int main()
      {
          /*  SELECT ' " ' FROM DUAL; */
      {
          struct ulpSqlstmt ulpSqlstmt;
          memset(&ulpSqlstmt, 0, sizeof(ulpSqlstmt));
          ulpSqlstmt.stmttype = 4;
          ulpSqlstmt.stmtname = NULL;
          ulpSqlstmt.ismt = 0;
          ulpSqlstmt.numofhostvar = 0;
          ulpSqlstmt.statusptr = NULL;
          ulpSqlstmt.errcodeptr = NULL;
          ulpSqlstmt.isatomic = 0;
          ulpSqlstmt.stmt = (char *)"SELECT ' \" ' FROM DUAL";
      ...
      }
      }
      ```

-   **Workaround**

-   **Changes**

    -   Performance view
    -   Property
    -   Compile Option
    -   Error Code

### BUG-50778<a name=bug-50778></a> Add exception handling to the getColumnCount function.

-   **module** : qp
-   **Category** : Fatal
-   **Reproducibility** : Rare

> -   **Description** : Altibase internally invokes the getColumnCount function to request column information during the Prepare process. There was an issue that the server was abnormally terminated because of incorrect data structure access in an exceptional situation. Altibase added exception handling to this situation.

Reviewer: ```suggestion
-   **Description** : Altibase internally invokes the getColumnCount function to request column information during the Prepare process. There was an issue that the server was abnormally terminated because of an invalid data structure access in an exceptional situation. Altibase added exception handling to this situation.
```

-   **How to reproduce this bug**
    -   **Reproduction conditions**
    -   **Actual Results**
    -   **Expected Results**

-   **Workaround**
-   **Changes**

    -   Performance view
    -   Property
    -   Compile Option
    -   Error Code

> ### BUG-50783<a name=bug-50783></a> Resolve an issue where an "Invalid use of host variables" error occurred during DDL on a table while executing a procedure.

Author: ```suggestion
### BUG-50783<a name=bug-50783></a> Resolve the issue where "ERR-3123B : Invalid use of host variables" occurred and a procedure was terminated when a DDL statement was executed on the table being referenced within the procedure.
```

-   **module** : qp

-   **Category** : Functional Error

-   **Reproducibility** : Always

> -   **Description** : Altibase resolved an issue where an  "ERR-3123B : Invalid use of host variables" error occurred during DDL on a table while executing a procedure.

Author: ```suggestion
-   **Description** : Altibase resolved the issue where "ERR-3123B : Invalid use of host variables" occurred and a procedure was terminated when a DDL statement was executed on the table being referenced within the procedure.
```

>     

Reviewer: 대략 이런뜻입니다. 
When a procedure is in execution and DDL is performed on the table used within the procedure's body, an 'Invalid use of host variable' error occurs.

    ```sql
    -- session 1
    CREATE TABLE T1 (I1 INTEGER);
    INSERT INTO T1 VALUES(1);
    CREATE OR REPLACE PROCEDURE PROC1 AS
    VAR1 INTEGER;
    VAR2 INTEGER;
    BEGIN
      VAR1 := 1;
      FOR I IN 1 .. 10 LOOP
      SELECT VAR1 INTO VAR2 FROM T1;
      PRINTLN(VAR2);
      SLEEP(2);
      VAR1 := VAR1 + 1;
      END LOOP;
    END;
    /
    EXEC PROC1;
    
    -- session 2
    ALTER TABLE T1 ADD (I2 INTEGER);
    ```

  - **Actual Results**

    ```sql
    iSQL> EXEC PROC1;
    1
    2
    3
    4
    [ERR-3123B : Invalid use of host variables
    0001 : select :B0 from T1;
                 ^  ^
    
    at "SYS.PROC1", line 7]
    ```

  -   **Expected Results**

      ```sql
      iSQL> EXEC PROC1;
      1
      2
      3
      4
      5
      6
      7
      8
      9
      10
      Execute success.
      ```

-   **Workaround**

-   **Changes**

    -   Performance view
    -   Property
    -   Compile Option
    -   Error Code

### BUG-50789<a name=bug-50789></a> Fix a problem where media recovery failed when additional checkpoint image files were added during the process.

-   **module** : sm
-   **Category** : Fatal
-   **Reproducibility** : Always

> -   **Description** : When additional checkpoint image files were added during the MEDIA RECOVERY process, the media recovery failed because it was not able to find LSNs of the added checkpoint image files. To resolve this, now Altibase records CreateLSN of the added file during the MEDIA RECOVERY process on its header.

Reviewer: ```suggestion
-   **Description** : When additional checkpoint image files were added during the MEDIA RECOVERY process, the media recovery failed because it could not find LSNs of the added checkpoint image files. To resolve this, now Altibase records CreateLSN on the file header that are added during the MEDIA RECOVERY process.
```

-   **How to reproduce this bug**
    -   **Reproduction conditions**

    -   **Actual Results**

    -   **Expected Results**

-   **Workaround**
-   **Changes**
    -   Performance view
    -   Property
    -   Compile Option
    -   Error Code

> ### BUG-50792<a name=bug-50792></a> Fix a logic error in calculating the maximum value of undo tablespace during snapshot specification.

Author: ```suggestion
### BUG-50792<a name=bug-50792></a> Fix a logic error in calculating the maximum value of undo tablespace when SNAPSHOT is set up.
```

> ### BUG-50792<a name=bug-50792></a> Fix a logic error in calculating the maximum value of undo tablespace during snapshot specification.

Reviewer: during snapshot specification -> with snapshot enabled. 가 어떤가요? 스냅샷이 설정된 상태에서

-   **module** : sm

-   **Category** : Functional Error

-   **Reproducibility** : Always

> -   **Description** : Altibase fixed a logic error in calculating the maximum value of undo tablespace during snapshot specification.

Author: ```suggestion
-   **Description** : Altibase fixed a logic error in calculating the maximum value of undo tablespace when SNAPSHOT is set up.
```
